core/modules/css/CssAccess.cc

```diff
@@ -0,0 +1,848 @@
+}
+
+void
+CssAccess::createDbLike(std::string const& dbName,
+                        std::string const& templateDbName) {
+    LOGF(_log, LOG_LVL_DEBUG, "createDbLike(%1%)" % dbName);
+
+    std::vector<std::string> subKeys{"partitioningId", "releaseStatus", "storageClass"};
+    auto dbMap = _getSubkeys(_prefix + "/DBS/" + templateDbName, subKeys);
+    if (dbMap.empty()) {
+        // nothing is found, check whether db exists
+        _assertDbExists(templateDbName);
+    }
+
+    // make new database with the copy of all parameters
+    std::string const dbKey = _prefix + "/DBS/" + dbName;
+    _storePacked(dbKey, dbMap);
+    _kvI->set(dbKey, "READY");
+}
+
+void
+CssAccess::dropDb(std::string const& dbName) {
+    LOGF(_log, LOG_LVL_DEBUG, "dropDb(%1%)" % dbName);
+
+    std::string key = _prefix + "/DBS/" + dbName;
+
+    // key is supposed to exist and key.json is optional
+    try {
+        LOGF(_log, LOG_LVL_DEBUG, "dropDb: try to delete packed key: %s.json" % key);
+        _kvI->deleteKey(key + ".json");
+    } catch (NoSuchKey const& exc) {
+        LOGF(_log, LOG_LVL_DEBUG, "dropDb: packed key is not found");
+    }
+    try {
+        LOGF(_log, LOG_LVL_DEBUG, "dropDb: try to delete regular key: %s" % key);
+        _kvI->deleteKey(key);
+    } catch (NoSuchKey const& exc) {
+        LOGF(_log, LOG_LVL_DEBUG, "dropDb: regular key is not found");
+        throw NoSuchDb(dbName);
+    }
+}
+
+std::vector<std::string>
+CssAccess::getTableNames(std::string const& dbName, bool readyOnly) const {
+    LOGF(_log, LOG_LVL_DEBUG, "getTableNames(%1%)" % dbName);
+
+    std::string key = _prefix + "/DBS/" + dbName + "/TABLES";
+    std::vector<std::string> names;
+    try {
+        names = _kvI->getChildren(key);
+    } catch (NoSuchKey const& exc) {
+        LOGF(_log, LOG_LVL_DEBUG, "getTableNames: key is not found: %s" % key);
+        throw NoSuchDb(dbName);
+    }
+
+    // remove names from packed keys
+    auto it = std::remove_if(names.begin(), names.end(),
+                             [](const std::string& name) {
+                                return name.compare(name.size()-5, name.size(), ".json") == 0; });
+    names.erase(it, names.end());
+
+    if (readyOnly and not names.empty()) {
+        // filter out names with status other than READY
+        auto const tableStatuses = _getSubkeys(key, names);
+        names.clear();
+        for (auto& kv: tableStatuses) {
+            if (kv.second == "READY") {
+                names.push_back(kv.first);
+            }
+        }
+    }
+    return names;
+}
+
+bool
+CssAccess::containsTable(std::string const& dbName, std::string const& tableName, bool readyOnly) const {
+    LOGF(_log, LOG_LVL_DEBUG, "containsTable(%1%, %2%)" % dbName % tableName);
+
+    std::string const key = _prefix + "/DBS/" + dbName + "/TABLES/" + tableName;
+    // If key is not there pretend that its value is not "READY"
+    std::string const val = _kvI->get(key, "DOES_NOT_EXIST");
```

Comment:
What mattes here is that key is "READY" or not, "DOES_NOT_EXIST" is the same as not "READY", I could use any other default value there.